File: algorithms/1154.day-of-the-year/solution.go
func dayOfYear(date string) int {
	year_month_day := make([]int, 3)
	for i, s := range strings.Split(date, "-") {
		year_month_day[i], _ = strconv.Atoi(s)
	}

	days := []int{
		0,
		31, // Jan.
		28, // Feb.
		31, // Mar.
		30, // Apr.
		31, // May.
		30, // June
		31, // July
		31, // Aug.
		30, // Sept.
		31, // Oct.
		30, // Nov.
		31, // Dec.
	}

	if leapYear(year_month_day[0]) {
		days[2] = 29
	}

	sum := 0
	for i := 0; i < year_month_day[1]; i++ {
		sum += days[i]
	}

	return sum + year_month_day[2]
}

func leapYear(year int) bool {
	return (year%4 == 0) && !(year%100 == 0) || (year%400 == 0)
}

File: algorithms/1154.day-of-the-year/solution.rs
impl Solution {
    pub fn day_of_year(date: String) -> i32 {
        let year_month_day: Vec<i32> = date.split("-").map(|s| s.parse::<i32>().unwrap()).collect();
        let mut days: Vec<i32> = vec![
            0, 31, // Jan.
            28, // Feb.
            31, // Mar.
            30, // Apr.
            31, // May.
            30, // June
            31, // July
            31, // Aug.
            30, // Sept.
            31, // Oct.
            30, // Nov.
            31, // Dec.
        ];

        if Self::leap_year(year_month_day[0]) {
            days[2] = 29;
        }

        let mut sum: i32 = 0;
        for i in 0..year_month_day[1] {
            sum += days[i as usize]
        }

        sum + year_month_day[2]
    }

    fn leap_year(year: i32) -> bool {
        (year % 4 == 0) && !(year % 100 == 0) || (year % 400 == 0)
    }
}
